src/precord.h

```diff
@@ -28,7 +31,53 @@
 **  Note that you still have to set the actual length once you have populated
 **  the record!
 */
-Obj NEW_PREC(UInt len);
+extern Obj NEW_PREC(UInt len);
+
+
+/****************************************************************************
+**
+*F  IS_PREC( <rec> ) . . . . . . . . .  check if <rec> is in plain record rep
+*/
+static inline Int IS_PREC(Obj rec)
+{
+    UInt tnum = TNUM_OBJ(rec);
+    return tnum == T_PREC || tnum == T_PREC+IMMUTABLE;
+}
+
+
+/****************************************************************************
+**
+*F  IS_PREC_OR_COMOBJ( <list> ) . . . . . . . . . . . . . check type of <rec>
+**
+**  Checks if this is 'PREC'-like.
+**  This function is used in a GAP_ASSERT checking if calling functions like
+**  SET_ELM_PREC is acceptable on an Obj.
+**
+**  Unlike IS_PREC, this function also accepts precs which are being copied
+**  (and hence have the COPYING flag set), as well as component objects
+**  (which have the same memory layout as precs), as the precs APIs using it
+**  for assertion checks are in practice invoked on such objects, too.
+*/
+static inline Int IS_PREC_OR_COMOBJ(Obj rec)
+{
+    UInt tnum = TNUM_OBJ(rec);
+    if (tnum > COPYING)
+        tnum -= COPYING;
+    return tnum == T_PREC || tnum == T_PREC+IMMUTABLE || tnum == T_COMOBJ;
+}
+
+
+/****************************************************************************
+**
+*F  CAPACITY_PREC(<list>) . . . . . . . . . . . .  capacity of a plain record
+**
+**  'CAPACITY_PREC' returns the maximum capacity of a PREC.
+**
+*/
+static inline Int CAPACITY_PREC(Obj rec)
+{
```

Comment:
Missing assert IS_PREC_OR_COMOBJ?

Reply:
I imitated `CAPACITY_PLIST` here, which also does not contain that check -- all callers to this function already know they are dealing with a prec, too.